gpkit/geometric_program.py

```diff
@@ -176,14 +195,46 @@ def solve(self, solver=None, verbosity=1, *args, **kwargs):
                             for p_i, m_is in enumerate(m_iss)])
         else:
             raise RuntimeWarning("The dual solution was not returned.")
-        result["sensitivities"]["monomials"] = nu
-        result["sensitivities"]["posynomials"] = la
 
+        ## Get sensitivities
+        result["sensitivities"] = {"constraints": {}}
+        # initialize the var_senss dict with the cost's constants...
+        var_senss = {var: sum([self.cost.exps[i][var]*nu[i] for i in locs])
+                     for (var, locs) in self.cost.varlocs.items()
+                     if (var in self.cost.varlocs
+                         and var not in self.posynomials[0].varlocs)}
+        var_senss = HashVector(var_senss)
+        for c_i, posy_idxs in enumerate(self.constr_idxs):
+            constr = self.constraints[c_i]
+            p_senss = [la[p_i] for p_i in posy_idxs]
+            m_sensss = [[nu[i] for i in self.m_idxs[p_i]] for p_i in posy_idxs]
+            constr_sens, p_var_senss = constr.sens_from_dual(p_senss, m_sensss)
+            # ...then add to it the constant sensitivities of each constraint
+            var_senss += p_var_senss
+            # also, add each constraint's sensitivities to the results
+            result["sensitivities"]["constraints"][str(constr)] = constr_sens
+        result["sensitivities"]["constants"] = KeyDict(var_senss)
```

Comment:
high level feeling comment (no action needed): maybe some of this sensitivity stuff could/should happen in `NomialData`?